File: app/src/main/java/com/reteno/sample/fragments/database/InteractionWriteDialogFragment.java
package com.reteno.sample.fragments.database;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.reteno.core.data.local.database.manager.RetenoDatabaseManagerInteraction;
import com.reteno.core.data.local.model.interaction.InteractionDb;
import com.reteno.core.data.local.model.interaction.InteractionStatusDb;
import com.reteno.core.util.UtilKt;
import com.reteno.sample.databinding.DialogDbWriteInteractionBinding;
import com.reteno.sample.util.Util;

public class InteractionWriteDialogFragment extends BaseDatabaseDialogFragment {

    private DialogDbWriteInteractionBinding binding;
    private RetenoDatabaseManagerInteraction databaseManager;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        databaseManager = serviceLocator.getRetenoDatabaseManagerInteractionProvider().get();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        databaseManager = serviceLocator.getRetenoDatabaseManagerInteractionProvider().get();
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        binding = DialogDbWriteInteractionBinding.inflate(getLayoutInflater());
        return new AlertDialog.Builder(requireActivity())
                .setView(binding.getRoot())
                .create();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    @Override
    public void onStart() {
        super.onStart();
        initListeners();
    }

    private void initListeners() {
        binding.btnSubmit.setOnClickListener(v -> {
            String interactionId = Util.getTextOrNull(binding.etInteractionId);
            String status = Util.getTextOrNull(binding.etStatus);
            String time = Util.getTextOrNull(binding.etTime);
            String token = Util.getTextOrNull(binding.etToken);

            if (UtilKt.allElementsNotNull(interactionId, status, time, token)) {
                InteractionDb interaction = new InteractionDb(
                        interactionId,
                        InteractionStatusDb.fromString(status),
                        time,
                        token
                );
                databaseManager.insertInteraction(interaction);
                Toast.makeText(this.getContext(), "Sent", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/reteno/sample/fragments/database/InteractionWriteDialogFragment.kt
package com.reteno.sample.fragments.database

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.reteno.core.data.local.database.manager.RetenoDatabaseManagerInteraction
import com.reteno.core.data.local.model.interaction.InteractionDb
import com.reteno.core.data.local.model.interaction.InteractionStatusDb.Companion.fromString
import com.reteno.core.util.allElementsNotNull
import com.reteno.sample.databinding.DialogDbWriteInteractionBinding
import com.reteno.sample.util.Util

class InteractionWriteDialogFragment : BaseDatabaseDialogFragment() {
    private var binding: DialogDbWriteInteractionBinding? = null
    private var databaseManager: RetenoDatabaseManagerInteraction? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        databaseManager = serviceLocator.retenoDatabaseManagerInteractionProvider.get()
    }

    override fun onDestroy() {
        super.onDestroy()
        databaseManager = serviceLocator.retenoDatabaseManagerInteractionProvider.get()
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        binding = DialogDbWriteInteractionBinding.inflate(
            layoutInflater
        )
        return AlertDialog.Builder(requireActivity())
            .setView(binding!!.getRoot())
            .create()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    override fun onStart() {
        super.onStart()
        initListeners()
    }

    private fun initListeners() {
        binding!!.btnSubmit.setOnClickListener {
            val interactionId = Util.getTextOrNull(binding!!.etInteractionId)
            val status = Util.getTextOrNull(binding!!.etStatus)
            val time = Util.getTextOrNull(binding!!.etTime)
            val token = Util.getTextOrNull(binding!!.etToken)
            val action = Util.getTextOrNull(binding!!.etAction)
            if (allElementsNotNull(
                    interactionId,
                    status,
                    time
                ) && (token != null || action != null)
            ) {
                val interaction = InteractionDb(
                    null,
                    interactionId.orEmpty(),
                    fromString(status),
                    time.orEmpty(),
                    token,
                    action
                )
                databaseManager!!.insertInteraction(interaction)
                Toast.makeText(this.context, "Sent", Toast.LENGTH_SHORT).show()
            }
        }
    }
}
